projects/frontegg-app/src/lib/frontegg-app-services.service.ts

```diff
@@ -0,0 +1,212 @@
+  private activateStateSubject$ = new BehaviorSubject<any>(null);
+  private apiTokensStateSubject$ = new BehaviorSubject<any>(null);
+  private forgotPasswordStateSubject$ = new BehaviorSubject<any>(null);
+  private loginStateSubject$ = new BehaviorSubject<any>(null);
+  private mfaStateSubject$ = new BehaviorSubject<any>(null);
+  private profileStateSubject$ = new BehaviorSubject<any>(null);
+  private rolesStateSubject$ = new BehaviorSubject<any>(null);
+  private routesStateSubject$ = new BehaviorSubject<any>(null);
+  private securityPolicyStateSubject$ = new BehaviorSubject<any>(null);
+  private signUpStateSubject$ = new BehaviorSubject<any>(null);
+  private socialLoginStateSubject$ = new BehaviorSubject<any>(null);
+  private ssoStateSubject$ = new BehaviorSubject<any>(null);
+  private teamStateSubject$ = new BehaviorSubject<any>(null);
+  private tenantsStateSubject$ = new BehaviorSubject<any>(null);
+  private userStateSubject$ = new BehaviorSubject<any>(null);
+
+  readonly acceptInvitationState$ = this.acceptInvitationStateSubject$.asObservable()
+  readonly accountSettingsState$ = this.accountSettingsStateSubject$.asObservable()
+  readonly activateState$ = this.activateStateSubject$.asObservable()
+  readonly apiTokensState$ = this.apiTokensStateSubject$.asObservable()
+  readonly forgotPasswordState$ = this.forgotPasswordStateSubject$.asObservable()
+  readonly loginState$ = this.loginStateSubject$.asObservable()
+  readonly mfaState$ = this.mfaStateSubject$.asObservable()
+  readonly profileState$ = this.profileStateSubject$.asObservable()
+  readonly rolesState$ = this.rolesStateSubject$.asObservable()
+  readonly routesState$ = this.routesStateSubject$.asObservable()
+  readonly securityPolicyState$ = this.securityPolicyStateSubject$.asObservable()
+  readonly signUpState$ = this.signUpStateSubject$.asObservable()
+  readonly socialLoginState$ = this.socialLoginStateSubject$.asObservable()
+  readonly ssoState$ = this.ssoStateSubject$.asObservable()
+  readonly teamState$ = this.teamStateSubject$.asObservable()
+  readonly tenantsState$ = this.tenantsStateSubject$.asObservable()
+  readonly userState$ = this.userStateSubject$.asObservable()
+
+
+  constructor(private fronteggAppService: FronteggAppService) {
+    // Memoized Auth State
+    this.fronteggAppService.fronteggAppAuthState$.pipe(filter((state) => !!state)).subscribe((authState) => {
+      if (!equal(this.acceptInvitationStateSubject$.getValue(), authState.acceptInvitationState)) {
+        this.acceptInvitationStateSubject$.next(authState.acceptInvitationState)
+      }
+      if (!equal(this.accountSettingsStateSubject$.getValue(), authState.accountSettingsState)) {
+        this.accountSettingsStateSubject$.next(authState.accountSettingsState)
+      }
+      if (!equal(this.activateStateSubject$.getValue(), authState.activateState)) {
+        this.activateStateSubject$.next(authState.activateState)
+      }
```

Comment:
what do you thing about:

```
if (!equal(this."acceptInvitationState"Subject$.getValue(), authState."acceptInvitationState")) {        
   this."acceptInvitationState"Subject$.next(authState."acceptInvitationState")      
}

```
you can create loops over object to prevent repeat if.

```
const authSubstates = ["acceptInvitationState", ...]

for(let subState in authSubstates){
  if(!equal(this[`${subState}Subject$`], authState[subState]){ 
    this[`${subState}Subject$`].next(authState[subState])
  } 
}
```

Reply:
It was a dry start =) 
Very good idea, thanks!